aws/src/test/java/org/apache/iceberg/aws/glue/DynamoLockManagerTest.java

```diff
@@ -0,0 +1,213 @@
+    properties.setGlueCatalogLockEnabled(true);
+    properties.setGlueCatalogLockTable(TABLE_NAME);
+    lockManager = new DynamoLockManager(dynamo, properties);
+  }
+
+  @After
+  public void after() throws Exception {
+    dynamo.deleteTable(DeleteTableRequest.builder()
+        .tableName(TABLE_NAME)
+        .build());
+    dynamoLocalProcess.destroy();
+    dynamoLocalProcess.waitFor();
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    Runtime.getRuntime().exec(String.format("rm %s.tar.gz", DYNAMO_LOCAL_DIR_NAME)).waitFor();
+    Runtime.getRuntime().exec(String.format("rm -rf %s", DYNAMO_LOCAL_DIR_NAME)).waitFor();
+  }
+
+  @Test
+  public void testTableCreation() {
+    try {
+      dynamo.describeTable(DescribeTableRequest.builder()
+          .tableName(TABLE_NAME)
+          .build());
+    } catch (Exception e) {
+      LOG.error("encountered failure", e);
+      Assert.fail("Should not throw exception when describing DynamoDB lock table");
+    }
+
+  }
+
+  @Test
+  public void testLock_singleProcess() {
+    boolean succeed = lockManager.tryLock("database.table", AwsProperties.GLUE_CATALOG_LOCK_EXPIRE_MS_DEFAULT);
+    Assert.assertTrue(succeed);
+    Map<String, AttributeValue> key = Maps.newHashMap();
+    key.put("lockId", AttributeValue.builder().s("database.table").build());
+    try {
+      dynamo.getItem(GetItemRequest.builder()
+          .tableName(TABLE_NAME)
+          .key(key)
+          .build());
+    } catch (Exception e) {
+      LOG.error("encountered failure", e);
+      Assert.fail("Should not throw exception when getting lock after insert");
+    }
+  }
+
+  @Test
+  public void testLock_multiProcess() {
+    List<Boolean> results = IntStream.range(0, 100).parallel()
+        .mapToObj(i -> lockManager.tryLock("database.table", AwsProperties.GLUE_CATALOG_LOCK_EXPIRE_MS_DEFAULT))
+        .collect(Collectors.toList());
+    Assert.assertEquals(1, results.stream().filter(s -> s).count());
+  }
+
+  @Test
+  public void testWaitLock_singleProcess() throws Exception {
+    boolean succeed = lockManager.tryLock("database.table",
+        AwsProperties.GLUE_CATALOG_LOCK_EXPIRE_MS_DEFAULT,
+        AwsProperties.GLUE_CATALOG_LOCK_WAIT_MS_DEFAULT);
+    Assert.assertTrue(succeed);
+    Map<String, AttributeValue> key = Maps.newHashMap();
+    key.put("lockId", AttributeValue.builder().s("database.table").build());
+    try {
+      dynamo.getItem(GetItemRequest.builder()
+          .tableName(TABLE_NAME)
+          .key(key)
+          .build());
+    } catch (Exception e) {
+      LOG.error("encountered failure", e);
+      Assert.fail("Should not throw exception when getting lock after insert");
+    }
+  }
+
+  @Test
+  public void testWaitLock_multiProcess_allSucceed() {
+    List<Boolean> results = IntStream.range(0, 100).parallel()
+        .mapToObj(i -> {
+          try {
+            return lockManager.tryLock("database.table", 10, 100000);
```

Comment:
I think what's happening here is that the attempts are deciding that others are expired. A comment would be helpful here.